ui/desktop/src/hooks/useToggleToolOutput.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { createContext, useContext, useState, useCallback, useEffect } from 'react';
+
+export interface HotkeyConfig {
+  key: string;
+  ctrl: boolean;
+  meta: boolean;
+  shift: boolean;
+  alt: boolean;
+}
+
+interface ToggleToolOutputContextType {
+  isExpandAll: boolean;
+  setExpandAll: (expand: boolean) => void;
+  toggleExpandAll: () => void;
+  hotkey: HotkeyConfig;
+  setHotkey: (hotkey: HotkeyConfig) => void;
+}
+
+const defaultHotkey: HotkeyConfig = {
+  key: 'e',
+  ctrl: true,
+  meta: false,
+  shift: false,
+  alt: false,
+};
+
+const ToggleToolOutputContext = createContext<ToggleToolOutputContextType | undefined>(undefined);
+
+export function ToggleToolOutputProvider({ children }: { children: React.ReactNode }) {
+  const [isExpandAll, setExpandAll] = useState(false);
+  const [hotkey, setHotkeyState] = useState<HotkeyConfig>(defaultHotkey);
+
+  const toggleExpandAll = useCallback(() => {
+    setExpandAll((prev) => !prev);
+  }, []);
+
+  const setExpandAllWrapper = useCallback((expand: boolean) => {
+    setExpandAll(expand);
+  }, []);
```

Comment:
so this means that once you hit this key, every time you come back to a conversation it will be expanded until you hit the key again?